tests/AssertAnnotationTest.php

```diff
@@ -437,6 +437,87 @@ function assertIntOrFoo($b) : void {
 
                     if (!is_int($a)) $a->bar();',
             ],
+            'unionOfStringTypes' => [
+                '<?php
+                /** @psalm-assert "self"|class-string $b */
+                function a(string $b): void {
+                    if ("" === $b) { throw new \Exception("");};
+                }
+
+                /** @psalm-assert-if-true "self"|class-string $b */
+                function atrue(string $b): bool {
+                    return $b === "";
+                }
+
+                /** @psalm-assert-if-false "self"|class-string $b */
+                function afalse(string $b): bool {
+                    return $b === "";
+                }
+
+                /** @param "self"|class-string $arg */
+                function check(string $arg): void {}
+
+                /** @var string $var */;
+                a($var);
+
+                /** @var string $varTrue */;
+                if (!atrue($varTrue)) {
+                    exit;
+                }
+
+                /** @var string $varFalse */;
+                if (afalse($varFalse)) {
+                    exit;
+                }
+
+                check($var);
+                check($varTrue);
+                check($varFalse);
+                ',
+            ],
+            'unionOfStringTypes2' => [
+                '<?php
+                    /** @psalm-assert "self"|class-string $b */
+                    function a(string $b): void {
+                        if ("" === $b) { throw new \Exception("");};
+                    }
+
+                    /** @psalm-assert-if-true "self"|class-string $b */
+                    function atrue(string $b): bool {
+                        return $b === "";
+                    }
+
+                    /** @psalm-assert-if-false "self"|class-string $b */
+                    function afalse(string $b): bool {
+                        return $b === "";
+                    }
+
+                    /** @param "self"|class-string $arg */
+                    function check(string $arg): void {}
+
+                    /** @var string $var */;
+                    a($var);
+
+                    /** @var string $varTrue */;
+                    if (!atrue($varTrue)) {
+                        exit;
+                    }
+
+                    /** @var string $varFalse */;
+                    if (afalse($varFalse)) {
+                        exit;
+                    }
+
+                    check($var);
+                    check($varTrue);
+                    check($varFalse);
+                    ',
+                [
+                    '$var' => 'class-string|"self"',
+                    '$varTrue' => 'class-string|"self"',
+                    '$varFalse' => 'class-string|"self"',
+                ],
+            ],
```

Comment:
Tests are identical, but second one checks for assertion statements and it looks like it doesn't work fine by some reason